server/src/main/java/io/crate/replication/logical/LogicalReplicationService.java

```diff
@@ -65,6 +131,46 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
     }
 
+    private void onChangedSubscriptions(@Nullable SubscriptionsMetadata prevSubscriptionsMetadata,
+                                        SubscriptionsMetadata newSubscriptionsMetadata) {
+        var oldSubscriptions = prevSubscriptionsMetadata == null ? null : prevSubscriptionsMetadata.subscription();
+        var newSubscriptions = newSubscriptionsMetadata.subscription();
+
+        for (var entry : newSubscriptions.entrySet()) {
+            var subscriptionName = entry.getKey();
+            if (oldSubscriptions == null || oldSubscriptions.get(subscriptionName) == null) {
+                assert repositoriesService != null : "RepositoriesService is not (yet) set";
```

Comment:
```suggestion
                assert repositoriesService != null : "RepositoriesService must be set immediately after LogicalReplicationService construction";
```

Instead of stating what went wrong I'd think of the assertion as a contract/assumption and the message as something that states this assumption.